src/main/java/org/sonar/plugins/csharp/SarifParser.java

```diff
@@ -44,33 +48,59 @@ public void parse(File file) {
     }
 
     JsonParser parser = new JsonParser();
-    JsonElement issues = parser.parse(contents).getAsJsonObject().get("issues");
-    if (issues != null) {
-      for (JsonElement issueElement : issues.getAsJsonArray()) {
-        JsonObject issue = issueElement.getAsJsonObject();
+    JsonObject root = parser.parse(contents).getAsJsonObject();
+    if (root.has("runLogs")) {
+      JsonElement runLogs = parser.parse(contents).getAsJsonObject().get("runLogs");
+      for (JsonElement runLogElement: runLogs.getAsJsonArray()) {
+        JsonObject runLog = runLogElement.getAsJsonObject();
+        JsonArray results = runLog.getAsJsonArray("results");
+        if (results != null) {
+          handleIssues(results, false, true);
+        }
+      }
+    } else if (root.has("issues")) {
+      JsonElement issues = parser.parse(contents).getAsJsonObject().get("issues");
+      handleIssues(issues.getAsJsonArray(), true, false);
+    }
+  }
+
+  private void handleIssues(JsonArray issues, boolean linesOffByOne, boolean convertUri) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "handleIssues" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)